config/config.go

```diff
@@ -476,14 +476,30 @@ func (t *TLSConfig) Validate() error {
 	return nil
 }
 
+type MetricsConfig struct {
```

Comment:
Please add a ```Validate()``` function to this config type. All config types have validation.

Reply:
since you want to use the same router, there is no port to validate .. so basically the only real config toggle is a bool (no_auth). Validation makes not much sense here ..
do you still want a validate func ?

btw, I removed the `disable metrics` toggle, mainly because in the router I don't have access to the config and I don't know if one would want to disable metrics. What do you think? shall we be able to disable metrics? 
viper would really be nice here :)

Reply:
> viper would really be nice here :)

We'll get there. Eventually :smile:. It's not worth the effort for now.

> do you still want a validate func ?

It's a good idea to have a ```Validate()``` func, even if it's a noop, **if we have a separate type** for a particular set of settings. If we embed the metrics auth in an existing type, and it's a bool, there is no need to validate it.

The idea is that we may extend a config at some point, and may forget to add a ```Validate()``` func later on.

> btw, I removed the `disable metrics` toggle, mainly because in the router I don't have access to the config and I don't know if one would want to disable metrics.

You can pass in a new param to ```NewAPIRouter()``` which includes whichever config you need. It didn't make sense to pass in a config to that func until now, but if that changes, I have nothing against adding a param that makes sense. 

I think it may be useful to have a toggle to disable metrics.

Reply:
.. I now pass the config into NewAPIRouter